packages/api/src/beacon/routes/beacon/pool.ts

```diff
@@ -1,70 +1,97 @@
-import {phase0, altair, capella, CommitteeIndex, Slot, ssz} from "@lodestar/types";
-import {ApiClientResponse} from "../../../interfaces.js";
-import {HttpStatusCode} from "../../../utils/client/httpStatusCode.js";
+/* eslint-disable @typescript-eslint/naming-convention */
+import {ValueOf} from "@chainsafe/ssz";
+import {phase0, capella, CommitteeIndex, Slot, ssz} from "@lodestar/types";
+import {Schema, Endpoint, RouteDefinitions} from "../../../utils/index.js";
 import {
-  RoutesData,
-  ReturnTypes,
   ArrayOf,
-  Schema,
-  reqOnlyBody,
-  ReqSerializers,
-  reqEmpty,
-  ReqEmpty,
-  ContainerData,
-} from "../../../utils/index.js";
+  EmptyArgs,
+  EmptyRequestCodec,
+  EmptyMeta,
+  EmptyMetaCodec,
+  EmptyRequest,
+  EmptyResponseCodec,
+  EmptyResponseData,
+} from "../../../utils/codecs.js";
 
 // See /packages/api/src/routes/index.ts for reasoning and instructions to add new routes
 
-export type AttestationFilters = {
-  slot: Slot;
-  committeeIndex: CommitteeIndex;
-};
+const AttestationListType = ArrayOf(ssz.phase0.Attestation);
+const AttesterSlashingListType = ArrayOf(ssz.phase0.AttesterSlashing);
+const ProposerSlashingListType = ArrayOf(ssz.phase0.ProposerSlashing);
+const SignedVoluntaryExitListType = ArrayOf(ssz.phase0.SignedVoluntaryExit);
+const SignedBLSToExecutionChangeListType = ArrayOf(ssz.capella.SignedBLSToExecutionChange);
+const SyncCommitteeMessageListType = ArrayOf(ssz.altair.SyncCommitteeMessage);
 
-export type Api = {
+type AttestationList = ValueOf<typeof AttestationListType>;
```

Comment:
This type is same as `phase0.Attestation[]` then why to declare with more complex patterns?

Reply:
I think the idea is to always match the ssz type by deriving the ts type from it. In this specific case, I think it's even better if we stick to `phase0.Attestation[]` as for electra we will have to change it to `allForks.Attestation[]` and construct the ssz type dynamically based on the fork.

I am honestly in favor of changing it as well as you suggested, and remove all -`List` suffixed type unless those are more broadly used (outside routes file).

Those were added by @wemeetagain, let's get his thoughts first

Reply:
> I think the idea is to always match the ssz type by deriving the ts type from it.

Yeah that was the idea. This is how its done in our types package, and my thought was that we may want to move all these types defined in the api over to our types package.

Reply:
> that we may want to move all these types defined in the api over to our types package.

Having beacon api only types defined in the api package seems like a good separation to me. I am still not sure about the `List` types as e.g. `AttestationList` really is `phase0.AttestationList`, so with current approach we would have to rework this to support electra apis.

This would be solved if we move `AttestationList` to types package (as suggested) but kinda bloats the package with types that are not part of consensus spec.